### PR TITLE
feat(esds-22): nuxt plugin support

### PR DESCRIPTION
**Pull request by RyanMulready**
### ❓ Type of change

- [X] 📖 Documentation (updates to the documentation or readme)
- [X] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [X] ✨ New feature (a non-breaking change that adds functionality)
- [X] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- renamed all icons to support auto import correctly; filename must match `name` and `id`
  - **this accounts for 73/126 of the file changes** 
- Removed aliases from our rollup config for `es-vue-base`; it breaks imports when using `src/*`
  - **this accounts for 11/126 of the file changes**
- Created new nuxt module to allow for auto importing of components based on usage
  - This is a now almost standard pattern in nuxt. its usually more proficient to let nuxt/webpack handle how components are bundled and this is easily accomplished using auto imported components.
- Enabled `modern` mode during `es-design-system` build to better test supporting modern mode
- Updates `es-design-system` to use new nuxt module and auto component imports
  - **this accounts for 32/126 of the file changes**
- Fixed an issue in `EsSlider` where change event triggered twice; introduced in `0.14.0`
- Fixed an issue in `EsFormRadioCards` where `value` was not defined as a prop; introduced in `0.15.0`
- Fixed an issue in `EsFormTextArea` where `hasMessage` was undefined; introduced in `0.13.0`
- Fixed typo in `es-form-msg`
- Fixed an issue in `es-form-radio-cards` where `key` could contain spaces

### 🥼 Testing

- Updated snapshots
- Manual testing nuxt plugin in `es-design-system`

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [X] I have updated the documentation accordingly.
- [X] I have documented testing approach

### CODE REVIEW

File: es-vue-base/README.md [hunk 1]
```diff
@@ -38,6 +39,19 @@ export default {
 </script>
 ```
 
+nuxt.config.js:
+
+```javascript
+    modules: [
+        ['@energysage/es-vue-base/nuxt', {
```
Comment (tomleo) [inline es-vue-base/README.md:46]: 👍

File: es-vue-base/nuxt/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    this.nuxt.hook('components:dirs', (dirs) => {
+        // Auto import components
+        if (options.components) {
+            dirs.push({
+                path: join(__dirname, '../src/lib-components'),
+                extensions: ['vue'],
+            });
+        }
+
+        // Auto import icons
```
Comment (tomleo) [inline es-vue-base/nuxt/index.js:19]: Some components include icons, would they break if not auto-importing icons? Should we just auto-import all components?

**Review by tomleo — APPROVED** [review]
Overall looks good 👍 

Do you think a pre-release to test this downstream is worth doing?

One small nit: in the future could we keep bulk-renaming / refactoring PRs separate from other changes?

**Comment by RyanMulready** [comment]
> One small nit: in the future could we keep bulk-renaming / refactoring PRs separate from other changes?

Point taken. I was pretty far down this path when I realized we needed to rename our icon files but in the future I'll try to keep them separated.

File: es-vue-base/nuxt/index.js (hunk 2 above)
Comment (RyanMulready) [reply es-vue-base/nuxt/index.js:19]: They will not break. The auto importing happens in side nuxt not vue so those icons used by components in `es-vue-base` keep their direct import statement.

**Review by ericdouglaspratt — COMMENTED** [review]
Nice - good to make it easier for teams to use ESDS. Curious to learn more about the efficiency of auto-importing vs. explicit imports!

File: es-design-system/pages/molecules/es-form-radio-cards.vue [hunk 3]
```diff
@@ -184,19 +174,19 @@ export default {
                 {
                     title: 'Back up power',
                     description: 'Run appliances during a power outage',
-                    icon: IconSettingsVertical,
+                    icon: 'IconSettingsVertical',
```
[the remark this answers is not in the document]
Comment (RyanMulready) [reply es-design-system/pages/molecules/es-form-radio-cards.vue:177]: Yea really nice pattern. We will use it again in the CMS project.

File: es-design-system/pages/molecules/es-form-radio-cards.vue [hunk 4]
```diff
@@ -22,8 +22,8 @@
                     v-model="form.propertyType"
                     label="What type of property do you want quotes for?">
                     <es-form-radio-card
-                        v-for="option in propertyTypeOptions"
-                        :key="option.value"
+                        v-for="(option, index) in propertyTypeOptions"
+                        :key="`property-${index}`"
```
[the remark this answers is not in the document]
Comment (RyanMulready) [reply es-design-system/pages/molecules/es-form-radio-cards.vue]: Dang, I've been using that pattern for awhile now. Good call out. Reverted in e17b893

**Pull request merged**